### ISSUE TITLE
fails at npm install

### ISSUE THREAD

**Issue by davechesebrough** [ponyfoo/ponyfoo#35]
Hi,

Every time I try to run a build of ponyfoo, in whatever environment (including debian) it fails to install phantomjs. The phantomjs library linked in your package.json references binaries that don't exist.

I've been trying to work around this for several weeks but ultimately have been unable to. I understand this is more of an issue with the dependency than with your code, but the steps listed in the README just do not work at all for me. I'm just wondering if anyone has run into this issue before. How do you have this working in your own dev environment?

Always the same error:
```
npm ERR! phantomjs@1.9.16 install: `node install.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the phantomjs@1.9.16 install script 'node install.js'.
npm ERR! This is most likely a problem with the phantomjs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node install.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls phantomjs
npm ERR! There is likely additional logging output above.
npm info preuninstall phantomjs@1.9.16
npm info uninstall phantomjs@1.9.16
npm info postuninstall phantomjs@1.9.16
```

I'd love to run my own instance of ponyfoo for a personal blog as it's clearly designed in a very smart way, but have been unable to do so so far. Thanks for putting this out there.

More detail about the failure in the dependency:

```
> phantomjs@1.9.16 install /opt/ponyfoo/node_modules/phantomjs
> node install.js

Downloading https://github.com/eugene1g/phantomjs/releases/download/2.0.0-bin/phantomjs-2.0.0-linux-x86_64.tar.bz2

root@b5e96748597f:/opt/ponyfoo# wget https://github.com/eugene1g/phantomjs/releases/download/2.0.0-bin/phantomjs-2.0.0-linux-x86_64.tar.bz2
--2016-02-20 06:31:49--  https://github.com/eugene1g/phantomjs/releases/download/2.0.0-bin/phantomjs-2.0.0-linux-x86_64.tar.bz2

Resolving github.com (github.com)... 192.30.252.128
Connecting to github.com (github.com)|192.30.252.128|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2016-02-20 06:31:50 ERROR 404: Not Found.
```

Got it working by using `"phantomjs": "2.1.3",` but now getting:

```
/opt/ponyfoo/.env.browser.json:1
module.exports=
               ^
ParseError: Unexpected token

...

SyntaxError: /opt/ponyfoo/.env.browser.json: Unexpected end of input
```

It's writing an empty `.env.browser.json` file on `npm start`.

**Comment by davechesebrough** [comment]
I feel quite silly - this particular error is resolved by `npm run preinstall`. Until now, I failed to notice you'd written that.

It looks like there is also an issue with fsevents@1.0.7 on my architecture (Fedora 23) hoping perhaps my note could be useful to others if I can resolve it.

**Comment by marvindanig** [comment]
It didn't solve for me. Fails here:

```
node-gyp rebuild

  CXX(target) Release/obj.target/contextify/src/contextify.o
/Users/marvindanig/Projects/blog/ponyfoo/node_modules/.bin/touch: line 1: syntax error near unexpected token `('
/Users/marvindanig/Projects/blog/ponyfoo/node_modules/.bin/touch: line 1: `var touch = require("../touch")'
make: *** [Release/obj.target/contextify/src/contextify.o] Error 2
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:100:13)
gyp ERR! stack     at ChildProcess.emit (events.js:185:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 15.2.0
gyp ERR! command "/usr/local/Cellar/node/5.6.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/marvindanig/Projects/blog/ponyfoo/node_modules/contextify
gyp ERR! node -v v5.6.0
gyp ERR! node-gyp -v v3.2.1
gyp ERR! not ok 

```

**Comment by davechesebrough** [comment]
I've experienced the Contextify error too, but not under Debian.

The reason the install script was writing an empty `.env.browser.json` is because `jq` was not installed. Installing `jq` in my container fixed the issue.

I got it running! I used a Docker container because I wanted to mimic Ponyfoo's target environment (Debian) as closely as possible without having to roll with a Debian physical box or VM. It needs some optimization, the fully compiled image is over 1GB.

I'm not sure Nicolás is interested in containerizing Ponyfoo, but I've got the fork I used on my GitHub page.
